api/src/pcapi/core/external_bookings/api.py

```diff
@@ -131,12 +131,15 @@ def cancel_event_ticket(
     try:
         parsed_response = pydantic_v1.parse_obj_as(serialize.ExternalEventCancelBookingResponse, response.json())
         if parsed_response.remainingQuantity:
-            new_quantity = (
-                parsed_response.remainingQuantity + stock.dnBookedQuantity - len(barcodes)
-                if is_booking_saved
-                else parsed_response.remainingQuantity + stock.dnBookedQuantity
-            )
-            stock.quantity = new_quantity
+            if parsed_response.remainingQuantity == "unlimited":
+                stock.quantity = None
+            else:
+                new_quantity = (
+                    parsed_response.remainingQuantity + stock.dnBookedQuantity - len(barcodes)
+                    if is_booking_saved
+                    else parsed_response.remainingQuantity + stock.dnBookedQuantity
+                )
```

Comment:
[style / question de goût] Pour ceux qui aiment moyennement les expressions ternaires sur plusieurs lignes : 

```suggestion
                new_quantity = parsed_response.remainingQuantity + stock.dnBookedQuantity
                if is_booking_saved:
                    new_quantity -= len(barcodes)
```